src/coreclr/debug/dbgutil/machoreader.cpp

```diff
@@ -255,22 +260,23 @@ MachOModule::ReadSymbolTable()
             return false;
         }
         _ASSERTE(m_symtabCommand != nullptr);
-        _ASSERTE(m_strtab == nullptr);
+        _ASSERTE(m_strtabAddress == 0);
 
-        m_reader.TraceVerbose("SYM: symoff %08x nsyms %d stroff %08x strsize %d\n",
+        m_reader.TraceVerbose("SYM: symoff %08x nsyms %d stroff %08x strsize %d iext %d next %d\n",
             m_symtabCommand->symoff,
             m_symtabCommand->nsyms,
             m_symtabCommand->stroff,
-            m_symtabCommand->strsize);
-
-        // Read symbol table. An array of "nlist" structs.
-        void* symtabAddress = GetAddressFromFileOffset(m_symtabCommand->symoff);
-        size_t symtabSize = sizeof(nlist_64) * m_symtabCommand->nsyms;
+            m_symtabCommand->strsize,
+            m_dysymtabCommand->iextdefsym,
+            m_dysymtabCommand->nextdefsym);
 
+        // Read the external symbol part of symbol table. An array of "nlist" structs.
+        void* symtabAddress = (void*)(GetAddressFromFileOffset(m_symtabCommand->symoff) + (m_dysymtabCommand->iextdefsym * sizeof(nlist_64)));
```

Comment:
nit: name should be something like extSymbolTableAddress, this doesn't point to the beginning of the symbol table.